### PR TITLE
Fix docs examples & typos

### PR DESCRIPTION
**Pull request by siahuat0727**
## What does this PR do?

Fixes docs typo

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LTGM !

**Review by ydcjeff — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Review by Borda — CHANGES_REQUESTED** [review]
mind check the failing docs examples?

**Review by awaelchli — APPROVED** [review]
@Borda this failing gpu test comes from master

File: docs/source/style_guide.rst [hunk 1]
```diff
@@ -44,7 +44,7 @@ Here's a LightningModule that defines a model:
             self.layer_2 = nn.Linear(...)
             self.layer_3 = nn.Linear(...)
```
Comment (Borda) [inline docs/source/style_guide.rst:45]:
```suggestion
            self.layer_2 = nn.Linear()
            self.layer_3 = nn.Linear()
```

File: docs/source/style_guide.rst [hunk 2]
```diff
@@ -85,10 +85,10 @@ Models defined as such leave you with many questions; what is coef_x? is it a st
 
 Instead, be explicit in your init
 
-.. code-block:: python
+.. testcode::
 
     class LitModel(pl.LightningModule):
-        def __init__(self, encoder: nn.Module, coeff_x: float = 0.2, lr: float = 1e-3)
+        def __init__(self, encoder: nn.Module, coef_x: float = 0.2, lr: float = 1e-3):
```
Comment (Borda) [inline docs/source/style_guide.rst:91]:
```suggestion
        def __init__(self, encoder: nn.Module, coef_x: float = 0.2, lr: float = 1e-3):
            pass
```

**Review by Borda — APPROVED** [review]

File: docs/source/style_guide.rst [hunk 3]
```diff
@@ -35,16 +35,16 @@ A LightningModule can define both a system and a model.
 
 Here's a LightningModule that defines a model:
 
-.. code-block:: python
+.. testcode::
 
     class LitModel(pl.LightningModule):
```
Comment (Borda) [inline docs/source/style_guide.rst:40]:
```suggestion
    class LitModel(LightningModule):
```

File: docs/source/style_guide.rst [hunk 4]
```diff
@@ -74,7 +74,7 @@ sensible defaults in the init so that the user doesn't have to guess.
 
 Here's an example where a user will have to go hunt through files to figure out how to init this LightningModule.
 
-.. code-block:: python
+.. testcode::
 
     class LitModel(pl.LightningModule):
```
Comment (Borda) [inline docs/source/style_guide.rst:79]:
```suggestion
    class LitModel(LightningModule):
```

File: docs/source/style_guide.rst [hunk 5]
```diff
@@ -35,16 +35,16 @@ A LightningModule can define both a system and a model.
...
 Here's a LightningModule that defines a model:
 
-.. code-block:: python
+.. testcode::
 
-    class LitModel(pl.LightningModule):
+    class LitModel(LightningModule):
         def __init__(self, num_layers: int = 3)
             super().__init__()
             self.layer_1 = nn.Linear(...)
```
Comment (Borda) [inline docs/source/style_guide.rst:43]:
```suggestion
            self.layer_1 = nn.Linear()
```
